uncontrast_studies/parsers/consciousness_measure_parser.py

```diff
@@ -23,119 +23,206 @@
+    resolved_consciousness_measures = []
+
+    cm_main_type_data = clean_list_from_data(item["Consciousness Measures Main type"])
+    cm_specific_type_data = clean_list_from_data(item["Consciousness Measures Specific type"])
+    cm_phase_data = clean_list_from_data(item["Consciousness Measures Phase"])
+    number_of_trials_objective_data = clean_list_from_data(
+        item["Consciousness Measures Number of trials for the objective measure"], integer=True
+    )
+    is_same_as_task_data = clean_list_from_data(
+        item["Consciousness Measures Is the measure taken from the same participants as the main task?"]
+    )
+    number_of_awareness_participants_data = clean_list_from_data(
+        item["Consciousness Measures Number of participants of the awareness test"], integer=True
+    )
+    is_above_chance_data = clean_list_from_data(item["Consciousness Measures Is the performance above chance?"])
+    is_trials_excluded_data = clean_list_from_data(
+        item["Consciousness Measures Were trials excluded from the analysis based on the measure?"]
+    )
+
+    is_singular_main_type = len(cm_main_type_data) == 1 and not cm_main_type_data == [""]
+    is_singular_specific_type = len(cm_specific_type_data) == 1
+    is_singular_phase = len(cm_phase_data) == 1 and not cm_phase_data == [""]
+    is_singular_number_of_trials = len(number_of_trials_objective_data) == 1
+    is_singular_number_of_participants = len(number_of_awareness_participants_data) == 1
+    is_singular_is_same_as_task = len(is_same_as_task_data) == 1 and not is_same_as_task_data == [""]
+    is_singular_is_above_chance = len(is_above_chance_data) == 1
+    is_singular_is_trials_excluded = len(is_trials_excluded_data) == 1 and not is_trials_excluded_data == [""]
+
+    is_multiple_main_type = len(cm_main_type_data) > 1
+    is_multiple_specific_type = len(cm_specific_type_data) > 1
+    is_multiple_phase = len(cm_phase_data) > 1
+    is_multiple_number_of_trials = len(number_of_trials_objective_data) > 1
+    is_multiple_number_of_participants = len(number_of_awareness_participants_data) > 1
+    is_multiple_is_same_as_task = len(is_same_as_task_data) > 1
+    is_multiple_is_above_chance = len(is_above_chance_data) > 1
+    is_multiple_is_trials_excluded = len(is_trials_excluded_data) > 1
+
+    is_singular_data = (
+        is_singular_main_type
+        and is_singular_specific_type
+        and is_singular_phase
+        and is_singular_number_of_trials
+        and is_singular_number_of_participants
+        and is_singular_is_same_as_task
+        and is_singular_is_above_chance
+        and is_singular_is_trials_excluded
+    )
+    is_multiple_data = (
+        is_multiple_main_type
+        and is_multiple_specific_type
+        and is_multiple_phase
+        and is_multiple_is_same_as_task
+        and is_multiple_is_trials_excluded
+        and is_multiple_is_above_chance
+    )
+
+    if is_singular_data:
```

Comment:
as it's only "and" conditions here - do those two options apply to all possible permutation? or should be some third option?

Reply:
yes and no :) later on you can see we're checking for multiple `number of trials` and multiple `number of participants in the awareness test`, that's because these 2 parameters don't always line up with is singular data or is multiple data. it was more clear and efficient for me to do this check independently. 
But in a larger sense: yes, these are the 2 options. anything else should be considered a mistake and raise an error.